crates/runtime/src/flight/do_put.rs

```diff
@@ -48,12 +48,19 @@ use crate::{
     timing::TimedStream,
 };
 
-use super::{Service, flightsql::prepared_statement_query, metrics};
+use super::{
+    Service, flightsql::prepared_statement_query, metrics, middleware::RateLimiterExtension,
+};
 
 pub(crate) async fn handle(
     flight_svc: &Service,
     request: Request<Streaming<FlightData>>,
 ) -> Result<Response<<Service as FlightService>::DoPutStream>, Status> {
+    let rate_limit_check_fn = request
```

Comment:
Does this need to be here? Or can be below on L85?

Reply:
It needs to be here because the request object gets consumed by the into_inner call on the next line